### PR TITLE
Initial import of gitlab rock-on

### PR DESCRIPTION
**Pull request by rcastberg**
Initiial import, added port 443 from the comments https://github.com/rockstor/rockstor-core/issues/1025. 443 doesn't respond and requires settings to be changed in the image. 

And i was unable to add two UI buttons, but i guess this is expected.

### CODE REVIEW

**Comment by fvollmer** [comment]
Is this line correct: `"min_size": 1073741824`? According to spec min_size is in KB. I think 1074 GB is a little bit to large.

File: gitlab-rc.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+						"protocol": "tcp",
+						"label": "SSH Gitlab port",
+						"description": "GitLab SSH port. Suggested default: 2222"
+					}
+				},
+			"volumes": {
+				"/var/opt/gitlab": {
+					"description": "Choose a Share for GitLab repositories. Eg: create a Share called gitlab-repos for this purpose alone.",
+					"label": "Repository Storage",
+					"min_size": 1073741824
```
Comment (fvollmer) [inline gitlab-rc.json]: Is this line correct (line 32) correct? ? According to spec min_size is in KB. I think 1074 GB (1073741824 KB) is too large.

**Comment by rcastbergw** [comment]
Didin't notice that it was in kb, changed and updated the log to have a minimum size. Left etc as figured it didn't need much space.

File: gitlab-rc.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+					"label": "Repository Storage",
+					"min_size": 1048576
+				},
+				"/etc/gitlab": {
+					"description": "Choose a Share for GitLab configuration. Eg: create a Share called gitlab-config for this purpose alone.",
+					"label": "Config Storage"
+				},
+				"/var/log/gitlab": {
+                                        "description": "Choose a Share for GitLab log files. Eg: create a Share called gitlab-logs for this purpose alone.",
+                                        "label": "Config logs"
```
Comment (fvollmer) [inline gitlab-rc.json]: Looks like a comma is missing in line 40 and the indentation is off (tab vs spaces).

**Comment by fvollmer** [comment]
Sorry to bother you again, but there is a very minor spelling mistake. A missing space before `Suggested default` at two lines.

**Comment by phillxnet** [comment]
@rcastberg I see you have addressed all the issue raised by @schakrava , thanks for that. In your opinion is this pull request now ready for merge. Thanks, I suspect this pr and your updates have somehow got overlooked in the recent flurry here. Thanks for your diligence and patience.
@fvollmer Thanks for helping with this one, I think it will be a real win for Rockstor to have GitLab as a default Rock-on.

**Comment by rcastberg** [comment]
So far i have not noticed any new issues. 
Yes i think that this is ready.

**Comment by schakrava** [comment]
I finally got around to this. Thanks @rcastberg for a nicely done pull request. It's working great. I am actually planning to move rockstor to self hosted gitlab using this rock-on. I am going to test it in the next few weeks. @phillxnet give it a try on your side too if you can.

**Comment by phillxnet** [comment]
@schakrava OK great. I've got to update my existing GitLab and hope to import what repos I've got in that instance into this Rock-on so thanks @rcastberg, @fvollmer, and @schakrava for pulling this together. Very useful.

**Comment by phillxnet** [comment]
@schakrava I've just added GitLab CE to the available rock-ons section in docs,
Pull request should be along in a few minutes.

**Pull request merged**